clients/src/main/java/org/apache/kafka/common/network/SslTransportLayer.java

```diff
@@ -828,8 +832,16 @@ private void handshakeFailure(SSLException sslException, boolean flush) throws I
 
         state = State.HANDSHAKE_FAILED;
         handshakeException = new SslAuthenticationException("SSL handshake failed", sslException);
-        if (!flush || flush(netWriteBuffer))
-            throw handshakeException;
+
+        // Attempt to flush any outgoing bytes. If this fails because remote end has closed the channel, log the
+        // exception, but continue to handle the handshake failure as an authentication exception.
+        try {
+            if (flush)
+                flush(netWriteBuffer);
```

Comment:
Previously, we would not throw the exception if `flush` returned `false`, but now we do. I assume that's intentional? May be worth explaining.

Reply:
@ijuma Thanks for the review. Sorry that was unintentional, updated code and comment.